backend/scripts/test-integration.sh

```diff
@@ -1,52 +1,132 @@
+TEST_PORT=15432  # Use non-standard port to avoid conflicts
+DB_USER="postgres"
+DB_PASSWORD="postgres"
+DB_NAME="mnemosyne_test"
+
+# Cleanup function
+cleanup() {
+    echo -e "\n${YELLOW}Cleaning up...${NC}"
+    docker stop $CONTAINER_NAME >/dev/null 2>&1 || true
+    docker rm $CONTAINER_NAME >/dev/null 2>&1 || true
+}
+
+# Set trap to cleanup on exit
+trap cleanup EXIT
 
 # Check if docker is installed
 if ! command -v docker &> /dev/null; then
-    echo "Docker is required but not installed. Please install Docker."
+    echo -e "${RED}Docker is required but not installed. Please install Docker.${NC}"
     exit 1
 fi
 
+# Check if container already exists and remove it
+if docker ps -a | grep -q $CONTAINER_NAME; then
+    echo "Removing existing test container..."
+    cleanup
+fi
+
 # Start PostgreSQL container
+echo -e "${BLUE}Starting PostgreSQL container on port $TEST_PORT...${NC}"
 docker run -d \
-  --name mnemosyne-test-db \
-  -e POSTGRES_USER=mnemosyne_test \
-  -e POSTGRES_PASSWORD=test_password \
-  -e POSTGRES_DB=mnemosyne_test \
-  -p 5432:5432 \
-  postgres:15
+  --name $CONTAINER_NAME \
+  -e POSTGRES_USER=$DB_USER \
+  -e POSTGRES_PASSWORD=$DB_PASSWORD \
+  -e POSTGRES_DB=$DB_NAME \
+  -p $TEST_PORT:5432 \
+  postgres:15-alpine >/dev/null
 
 # Wait for PostgreSQL to be ready
-echo "Waiting for PostgreSQL to be ready..."
+echo -n "Waiting for PostgreSQL to be ready"
 for i in {1..30}; do
-  if docker exec mnemosyne-test-db pg_isready -U mnemosyne_test > /dev/null 2>&1; then
-    echo "PostgreSQL is ready!"
+  if docker exec $CONTAINER_NAME pg_isready -U $DB_USER >/dev/null 2>&1; then
+    echo -e " ${GREEN}✓${NC}"
     break
   fi
   echo -n "."
   sleep 1
+  if [ $i -eq 30 ]; then
+    echo -e " ${RED}✗${NC}"
+    echo -e "${RED}PostgreSQL failed to start within 30 seconds${NC}"
+    exit 1
+  fi
 done
 
-# Set environment variables
-export DB_HOST=localhost
-export DB_PORT=5432
-export DB_USER=mnemosyne_test
-export DB_PASSWORD=test_password
-export DB_NAME=mnemosyne_test
-export DB_SSLMODE=disable
+# Create test database config
+export TEST_DB_HOST=localhost
+export TEST_DB_PORT=$TEST_PORT
+export TEST_DB_USER=$DB_USER
+export TEST_DB_PASSWORD=$DB_PASSWORD
+export TEST_DB_NAME=$DB_NAME
+export TEST_DB_SSLMODE=disable
+
+# Run schema setup
+echo -e "${BLUE}Setting up database schema...${NC}"
+if [ -f internal/db/schema.sql ]; then
+    PGPASSWORD=$DB_PASSWORD psql -h localhost -p $TEST_PORT -U $DB_USER -d $DB_NAME -f internal/db/schema.sql >/dev/null 2>&1
+    echo -e "${GREEN}✓ Schema created${NC}"
+fi
+
+# Run migrations
+if [ -d internal/db/migrations ]; then
+    for migration in internal/db/migrations/*.sql; do
+        if [ -f "$migration" ]; then
+            echo -n "Applying $(basename $migration)... "
+            PGPASSWORD=$DB_PASSWORD psql -h localhost -p $TEST_PORT -U $DB_USER -d $DB_NAME -f "$migration" >/dev/null 2>&1
+            echo -e "${GREEN}✓${NC}"
+        fi
+    done
+fi
```

Comment:
The test pattern matching with grep could be improved. Consider using Go build tags for more explicit integration test marking.